### ISSUE TITLE
When renaming a language, existing content tagged with old language code will become zombie content

### ISSUE THREAD

**Issue by sknorr** [gaois/terminologue#58]
I changed the language code for English from `en` to `en-us`. I thought what would happen is that content previously part of `en` would now be `en-us` content. That's not what happened.

Instead, the old content (and localized labels etc.) wound up being inaccessible by the UI. It would be very bad, if things were now deleted. Luckily though, the old content is not deleted and changing the language code back to `en` made the content reappear.
Still -- this seems suboptimal.

**Comment by michmech** [comment]
Yes, I can see how silly this is. This is a consequence of the fact that the back-end storage isn't a proper relational database. The back-end storage is basically just a document database for blobs of JSON, and there's nothing there to take care of referential integrity automatically.

This kind of thing wasn't an issue when we were planning Terminologue originally. Terminologue was designed for the kind of termbases you plan very carefully in advance. It didn't seem necessary to worry about people wanting to rename language codes half-way through. But now that Terminologue has come to be adopted by a move diverse user community with more varied use cases, this assumption is no longer valid.

A workaround would be to prohibit people from changing the language code once we know that the code is being used in at least one term. Not great either, I know, but at the very least it would save people from panic attacks thinking all their content has been deleted.

**Comment by sknorr** [comment]
Thanks for the explanation! I'd already wondered a bit what was behind the "edit source code" fields that pervade the UI (like for manual user setup). Now that seems clearer to me.

As to the practicalities in our case, this is not a showstopper for us, as we're still very much in the planning phase and updating our existing TBX file to the right language codes is still a possibility. It's just helpful to know that we need to get it right, now. Or do it from another export later on and suffer a bit of a history reset.
